src/hapi.erl

```diff
@@ -144,11 +146,12 @@ req(Method, URI0, Opts) when is_map(URI0), map_size(URI0) > 0 ->
     Host = maps:get(host, URI),
     Port = maps:get(port, URI, 80),
     Path = maps:get(path, URI, ""),
+    Scheme = list_to_existing_atom(maps:get(scheme, URI)),
```

Comment:
According to https://www.erlang.org/doc/man/uri_string.html#type-uri_map spec the `scheme` is `unicode:chardata()` meaning it can be a _binary_ as well. Since we only support _http_ and _https_ here, I would map the scheme explicitly:
```erlang
format_scheme("http") -> http;
format_scheme(<<"http">>) -> http;
format_scheme("https") -> https;
format_scheme(<<"https">>) -> https.
```

Reply:
Done